File: Weatley.Web/src/app/core/entities/booking.ts
import { Customer } from './customer';
import { Room } from './room';
import { BookedRoom } from './bookedRoom';

export class Booking {
	id: string;
	startingDate: Date;
	endDate: Date;
	comment: string;
	price: number;
	customer: Customer;
	bookedRooms: BookedRoom[];
	rooms: string;

	constructor(values: Object = {}) {
		Object.assign(this, values);
	}
}

File: Weatley.Backend/Weatley.Model/Entities/Booking.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Weatley.Model.Entities
{
    public class Booking : IEntityBase
    {
        public Guid Id { get; set; }
        public DateTime StartingDate { get; set; }
        public DateTime EndDate { get; set; }
        public string Comment { get; set; }
        public double Price { get; set; }

        //Relation with Room
        public ICollection<BookedRoom> BookedRooms { get; set; }
    }
}
